gems/pending/VMwareWebService/VimService.rb

```diff
@@ -892,8 +902,19 @@ def retrievePropertiesEx(propCol, specSet, max_objects = nil)
   end
 
   def retrievePropertiesCompat(propCol, specSet, max_objects = nil)
+    oc = VimArray.new('ArrayOfObjectContent')
+
     rv = retrievePropertiesEx(propCol, specSet, max_objects)
-    rv ? rv['objects'] : []
+    if rv
+      oc.concat(rv['objects'])
+
+      while rv && rv['token']
+        rv = continueRetrievePropertiesEx(propCol, rv['token'])
+        oc.concat(rv['objects']) unless rv.nil?
```

Comment:
That was `waitForUpdatesEx` in `monitorUpdatesInitial`, we can do the same thing here too if we have the callers pass a block.